phoenix-core/src/main/java/org/apache/phoenix/query/ConnectionQueryServicesImpl.java

```diff
@@ -1553,11 +1543,20 @@ public GetVersionResponse call(MetaDataService instance)
                 long serverJarVersion = versionResponse.getVersion();
                 isTableNamespaceMappingEnabled |= MetaDataUtil.decodeTableNamespaceMappingEnabled(serverJarVersion);
 
-                if (!isCompatible(serverJarVersion)) {
-                    isIncompatible = true;
-                    HRegionLocation name = regionMap.get(result.getKey());
-                    buf.append(name);
-                    buf.append(';');
+                MetaDataUtil.ClientServerCompatibility compatibility = MetaDataUtil.areClientAndServerCompatible(serverJarVersion);
+                if (!compatibility.getIsCompatible()) {
+                    if (compatibility.getErrorCode() == SQLExceptionCode.OUTDATED_JARS.getErrorCode()) {
+                        HRegionLocation name = regionMap.get(result.getKey());
+                        errorMessage.append("Newer Phoenix clients can't communicate with older "
+                                + "Phoenix servers. " + getClientandServerVersions(serverJarVersion)
+                                + " The following servers require an updated "
+                                + QueryConstants.DEFAULT_COPROCESS_JAR_NAME
+                                + " to be put in the classpath of HBase: ");
+                        errorMessage.append(name);
+                        errorMessage.append(';');
+                    } else {
```

Comment:
Make this more specific with an `else if` rather than an `else`? Otherwise this breaks if tomorrow we have other cases for incompatibility.